### DISCUSSION TITLE
Reusing the graph rendering of Ariflow

### DISCUSSION THREAD

**Discussion by drorata** [apache/airflow#14133] [Q&A]
I would like to reuse the graph rendering Airflow is using. I tried to dig into the code a little, but so far with limited success; it seems like the rendering happens outside of Python, if I get it right and I'm getting a little lost there.

I would like to specify the graph that I have (with `networkx` maybe) and then render it using the "engine" used by Airflow. Ideas how to go about this would be very welcomed! Thanks!

**Comment by kaxil** [comment]
https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#show -- is this useful?

**Reply by kaxil** [reply]
More explanation is here too: 
https://airflow.apache.org/docs/apache-airflow/stable/usage-cli.html#exporting-dag-structure-as-an-image

```shell
$ airflow dags show example_complex --save example_complex.png
```

![image](https://user-images.githubusercontent.com/8811558/107252869-ac329200-6a2d-11eb-8715-d3c0e3e975b7.png)

**Reply by drorata** [reply]
As far as I can tell, not really. I want to reuse the rendering engine for other graphs (not Airflow related at all).

**Reply by kaxil** [reply]
yeah the current rendering happens in JS here - https://github.com/apache/airflow/blob/028612123d9be45fcbfa31f28d241f7c2226d33e/airflow/www/templates/airflow/graph.html#L126-L751

**Reply by drorata** [reply]
How does the input of this rendering look like? Where can I find its sample? Furthermore, can you please point me to the part where the graph's data is loaded?

**Reply by kaxil** [reply]
It is passed on from https://github.com/apache/airflow/blob/f180fa13bf2a0ffa31b30bb21468510fe8a20131/airflow/www/views.py#L2012-L2101

**Reply by drorata** [reply]
Thanks a lot! I'm afraid it is way more complex than I expected and I don't know how to reuse it :(
